src/commands/deployment/execute.rs

```diff
@@ -379,113 +385,74 @@ pub async fn handle_execute(
         let progress_bar = new_spinner_progress_bar();
         progress_bar.set_message("Fetching available build artifacts ...");
 
-        let cd_pipeline_data = client
-            .fetch_cd_pipeline(
+        let cd_pipeline_data: Option<CdPipelineWithBuilds>;
+        let mut pipeline_type = PipelineType::Github;
+
+        let github_cd_pipeline = get_github_cd_pipeline(
+            &mut client,
+            &current_application,
+            &selected_namespace.to_lowercase(),
+            &selected_version.to_lowercase(),
+        )
+        .await;
+
+        if github_cd_pipeline.is_ok() {
+            let github_cd_pipeline_cd = github_cd_pipeline.unwrap();
```

Comment:
while the code is valid, but I would prefer to use `if let Ok(github_cd_pipeline_cd) = github_cd_pipeline { ... }` so we don't have so much unwrap.